core/src/main/java/org/apache/spark/api/java/function/DoubleFunction.java

```diff
@@ -23,5 +23,5 @@
  *  A function that returns Doubles, and can be used to construct DoubleRDDs.
  */
 public interface DoubleFunction<T> extends Serializable {
-  public double call(T t) throws Exception;
+  double call(T t) throws Exception;
```

Comment:
For those looking at this change, FYI most of the apparent visibility changes in this PR are not changes. For example, `enum` constructor visibility is irrelevant, inner `enum`s and `interface`s are inherently `static`, a `public` constructor in a package-private class is effectively package-private, and interface methods are inherently `public` for example.

I support all the changes here in that it's how I write Java. I'm not sure how people feel about some of the changes here though. On the one hand, I'd definitely remove `public` from a `public` constructor in a package-private class since the visibility modifier is misleading. For interface methods, I'm of two minds, since `public` is suspiciously redundant, but, it also correctly states its visibility.

I personally would still make all these changes but some of these changes are up for debate and wondering if others have an opinion.